File: leetcode-5/answer.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        leng, head, tail = 0, 0, 0
        for i in range(len(s)):
            j, k = i, i
            while j-1 >= 0 and k+1 < len(s) and s[j-1] == s[k+1]:
                j -= 1
                k += 1
            if k-j > leng:
                leng = k-j
                head = j
                tail = k
            if i+1 < len(s) and s[i] == s[i+1]:
                j, k = i, i+1
                while j-1 >= 0 and k+1 < len(s) and s[j-1] == s[k+1]:
                    j -= 1
                    k += 1
                if k-j > leng:
                    leng = k-j
                    head = j
                    tail = k
        return s[head:tail+1]

File: leetcode-5/answer.cpp
#include <string>
#include <iostream>

using namespace std;

class Solution
{
public:
    string longestPalindrome(string s)
    {
        string::difference_type len = 0;
        string::iterator head = s.begin(), tail = s.begin();
        for (auto i = s.begin(); i != s.end(); ++i)
        {
            auto j = i, k = i;
            while (j - 1 >= s.begin() && k + 1 < s.end() && *(j - 1) == *(k + 1))
            {
                --j;
                ++k;
            }
            if (len < k - j)
            {
                len = k - j;
                head = j;
                tail = k;
            }
            if (i + 1 < s.end() && *i == *(i + 1))
            {
                j = i;
                k = i + 1;
                while (j - 1 >= s.begin() && k + 1 < s.end() && *(j - 1) == *(k + 1))
                {
                    --j;
                    ++k;
                }
                if (len < k - j)
                {
                    len = k - j;
                    head = j;
                    tail = k;
                }
            }
        }
        return s.substr(head - s.begin(), len + 1);
    }
};